fetch_telemetry.py

```diff
@@ -48,97 +51,129 @@
     #"include_calcs": True, #include calc-type blobs in spacecraft blob queries
 }
 
-def fetch_telemetry(stop= None):
-    fetch_result = get_CSH_blobs(stop)
+#
+#--- For selecting msid values from previous blobs for limit checking
+#
+COMP_LIM_SELECTION = ['COTLRDSF', 'COSCS128S', 'COSCS129S', 'COSCS130S', 'COTLRDSF', 'COBSRQID', '3TSCPOS', 'AOPCADMD', 'COBSRQID', 'CORADMEN']
+
+def fetch_telemetry(stop = None):
+    fetch_result, stop = get_blobs(stop)
 #
 #--- If the fetch result contains no blobs, then we are out of comm. 
 #
     if len(fetch_result['blobs']) > 0:
-        formatted_data = format_result(fetch_result)
+        latest_data_points = keep_latest_data_point(fetch_result)
 
-        unit_converted_data = unit_conversion(formatted_data)
+        unit_converted_data = unit_conversion(latest_data_points)
 
         pseudo_update_data = generate_psuedo_msids(unit_converted_data)
+#
+#--- Pull the last known values of other msids used in comparing limit values
+#
+        if os.path.isfile(f"{HOUSE_KEEPING}/comp_limit_values.json"):
+            with open(f"{HOUSE_KEEPING}/comp_limit_values.json") as f:
+                comp_lim_values = json.load(f)
+        else:
+            fetch_result = maude.get_msids(msids = COMP_LIM_SELECTION, stop = stop, nearest = True)
+            comp_lim_values = {}
+            for entry in fetch_result['data']:
+                comp_lim_values[entry['msid']] = str(entry['values'][-1])
+#
+#--- If the current blob update contains data from COMP_LIM_SELECTION, then update
+#
+        for msid in COMP_LIM_SELECTION:
+            x = pseudo_update_data.get(msid)
+            if x is not None:
+                comp_lim_values[msid] = str(x['value'])
+        
+        with open(f"{HOUSE_KEEPING}/comp_limit_values.json","w") as f:
+            json.dump(comp_lim_values,f,indent = 4)
```

Comment:
The COMP_LIM_SELECTION msids are used in the check status edge cases. With each data pull, we check a file kept in housekeeping for the value of these msids. We update these comparison values if the latest telemetry fetch contains update data, and then use them to update the status of the rest of the msids.